src/ess/reduce/nexus/workflow.py

```diff
@@ -491,6 +492,34 @@ def get_calibrated_monitor(
     )
 
 
+def assemble_monitor_beamline(
+    monitor: CalibratedMonitor[RunType, MonitorType],
+    source_position: Position[snx.NXsource, RunType],
+    gravity: GravityVector,
+) -> CalibratedMonitorBeamline[RunType, MonitorType]:
+    """
+    Add beamline information (gravity vector, source- and sample-position) to monitor.
+
+    This is performed separately and after :py:func:`get_calibrated_monitor` to avoid
+    as false dependency of, e.g., the reshaped detector numbers on the sample position.
+    The latter can change during a run, e.g., for a rotating sample. The detector
+    numbers might be used, e.g., to mask certain detector pixels, and should not depend
+    on the sample position.
```

Comment:
This does not apply here? Is there thus any reason to not perform this in `get_calibrated_monitor`? Would avoid the extra type and the "need" to rename `CalibratedBeamline`.